### PR TITLE
Add a test case for duplicate engine loading

### PR DESCRIPTION
**Pull request by bernd-edlinger**
This is just the test case from #17073 ported to 1.1.1,
The fixing part of #17073 cherry-picks cleanly, just the test
case had a non-tivial merge conflict.
It is expected that the CI will fail, that will be fixed when #17073
is merged.

### CODE REVIEW

**Comment by bernd-edlinger** [comment]
@t8m @beldmit this is the test case for 1.1.1, please approve this PR too?

**Comment by mattcaswell** [comment]
The CI failure looks relevant

**Comment by bernd-edlinger** [comment]
Yeah, that's because this is meant to be merged after #17073 part 1.

**Review by beldmit — APPROVED** [review]
LGTM, tests should be rerun after merging #17073

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
Merging needs to wait after the #17073 is merged to 1.1.1.

**Comment by xnox** [comment]
> Merging needs to wait after the #17073 is merged to 1.1.1.

has been merged.

**Comment by beldmit** [comment]
Close/open to rerun tests

**Comment by bernd-edlinger** [comment]
Merged to 1.1.1 as 6d022b0.  Thanks!

**Pull request closed without merging**
